llvm/lib/Target/AMDGPU/SIRegisterInfo.cpp

```diff
@@ -1657,8 +1657,9 @@ void SIRegisterInfo::buildSpillLoadStore(
     } else {
       MIB.addReg(SOffset, SOffsetRegState);
     }
-    MIB.addImm(Offset + RegOffset)
-       .addImm(0); // cpol
+
+    int64_t CPol = AMDGPU::isGFX12Plus(ST) && LastUse ? AMDGPU::CPol::TH_LU : 0;
```

Comment:
You don't really need to check isGFX12Plus here. It could be an assert. Pre-GFX12, last_use should always be 0.